kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -78,34 +87,48 @@
   // during read operations.
   private final Map<ConsumerInstanceId, KafkaConsumerState> consumers =
       new HashMap<ConsumerInstanceId, KafkaConsumerState>();
-  // Read operations are common and there may be many concurrently, so they are farmed out to
-  // worker threads that can efficiently interleave the operations. Currently we're just using a
-  // simple round-robin scheduler.
-  private final List<KafkaConsumerWorker> workers;
-  private final AtomicInteger nextWorker;
-  // A few other operations, like commit offsets and closing a consumer can't be interleaved, but
-  // they're also comparatively rare. These are executed serially in a dedicated thread.
+  // All kind of operations, like reading records, committing offsets and closing a consumer
+  // are executed separately in dedicated threads via a cached thread pool.
   private final ExecutorService executor;
   private KafkaConsumerFactory consumerFactory;
   private final PriorityQueue<KafkaConsumerState> consumersByExpiration =
       new PriorityQueue<KafkaConsumerState>();
+  private final DelayQueue<RunnableReadTask> delayedReadTasks = new DelayQueue<>();
+  private final ConcurrentHashMap<RunnableReadTask, Boolean> delayedTasks =
```

Comment:
I'm not sure I understand why we need this. The state machine as I understand is the following:

1. Task creation
2. Submission to executor [rejection -> delay queue]
3. Either:
a. Execution [incomplete read -> delay queue]
b Delay queue poll and resubmission
4. Task completion

Is there anywhere that we might submit the task twice to the queue?